apps/rest/rest.app.js

```diff
@@ -0,0 +1,322 @@
+const REST = 'REST';
+const DOIT = 'DOIT';
+
+let mode = MAIN_MENU;
+let lastRender = MAIN_MENU;
+
+function setMode(newMode){
+  mode = newMode;
+}
+
+function getImg() {
+  return require("heatshrink").decompress(atob("rFYwcBpMkyQCB6QFDmnStsk6dpmmatO2AoMm7VpkmapMm6Vp02TEAmSCIIFB2mbEYPbtu07VJmwFCzYRD0gdB0gmBEAgCCtoOBtIOBIIPTpo1BHwJQCAQMmydNI4RBFLIILDmnaps2L4Om7ZEBI4IgCAQNN0g+GJQKJDKwIaB0iJCJQQmBCgWmHAIdEHYKnFDQSbBkBcE0wOBFgImBSoMmQZJTE6VAbYMJPQRHBDQKMBmmTtoUCEBPSJQT8CgKPCcAJQEIILFHMohxDEAUANwZ9E0wdBUhDLGyAgDO4LIByYOBAQLpEL45KEm2AQIMkwEEYQZTB7Vt23TC4wCHCgOAgRUBEAL+CzVtkwRCHw4CJEANNm2QggXEX4jpBIJgCBgESOoKHB6RiByYCBDQSGCMoIdJHAQgCkmCgALCZALpCd4RiNYoKkCkESpC8CEYm2QByDDgEBkETpBWDtukKYZBOHAKkBgIGBIIRNC0wFEIKCDCyVEBASbLAReQEAXSghKCzQ7BQYIgUoAGBEARuDIKmSgAAByAgFASwgCgALFmikUEBRBYgggcwBBDtDrDASwfDgFIgAgYkAfDgVAgEJECw6BAAcSEAKGXDIUAhEgZIcEYS4ABAwwgUyAgFAwjIUDIifBdQggUDIkBZIjKBECYZEAA4gSHQogoRYIgQD5gghgIgQpAg/QeAgRQcNAggeLECQDBwAgryIgTxAgKwAgQpQgKgMhkmQIKcIIJEgEA+kEBNApMgdJBhBgkQIKFCpMAEBUAMQ+aIJUioAgKIItpIJkCEBEAIJIgKhIgMyRBFmikLMRMAgkEEAmTUhogRARlAhIggkAgLUiNIpMgD5AgWXQIgcpMJED8BEBmAED0kwIgRkAgLkAgSkMkwAhKxIgRkgggXIIcFgIEDaYIgRwggGgBKDECcEyVAgEQEIkSpIgUgADCQwzSBEC0gD4pBBkQdQDgYgIBAIgVHAJFBcYgMBgQgUPQIgFFINIBQQgQTYYgfXQIgFFYggPGgIVCgmQDogFCECr8CII4KCECUBED4AKFYQgOoAYFggIGEC4XDEDgLDkAgVD4kCBYgKEECsSBYmAEDILFEEGQEBYA=="));
+}
+
+const onTouchPerQuadrantPerMode = {
+  // mode -> [[nextMode on touch, custom function], ... for all quadrants]
+  MAIN_MENU: [
+    [SET_REST, null], [SET_SETS, null],
+    [DOIT, null], [EXERCISE, null]
+  ],
+  SET_REST: [
+    [MAIN_MENU, Bangle.buzz], [null, null],
+    [null, () => {
+      restSeconds = Math.min(120, Math.max(0, restSeconds - 15));
+      Bangle.buzz(100);
+    }],
+    [null, () => {
+      restSeconds = Math.min(120, Math.max(0, restSeconds + 15));
+      Bangle.buzz(100);
+    }],
+  ],
+  SET_SETS: [
+    [null, null], [MAIN_MENU, Bangle.buzz],
+    [null, () => {
+      setsCount = Math.min(15, Math.max(0, setsCount - 1));
+      Bangle.buzz(100);
+    }],
+    [null, () => {
+      setsCount = Math.min(15, Math.max(0, setsCount + 1));
+      Bangle.buzz(100);
+    }],
+  ],
+  EXERCISE: [
+      [null, null], [null, null],
+      [null, null], [null, null],
+  ],
+  REST: [
+      [null, null], [null, null],
+      [null, null], [null, null],
+  ]
+}
+
+const drawFuncPerMode = {
+  MAIN_MENU: drawMainMenu,
+  SET_REST: drawSetRest,
+  SET_SETS: drawSetSets,
+  EXERCISE: drawExercise,
+  REST: drawRest,
+  DOIT: drawDoIt,
+}
+
+function redrawApp(){
+  g.setBgColor(-1).clear();
+  Bangle.drawWidgets();
+  drawFuncPerMode[mode]();
+  lastRender = mode;
+}
+
+function buttonPress () {
+  if (mode === EXERCISE) {
+    setMode(REST);
+    restUntil = Date.now() + (restSeconds * 1000);
+    redrawApp();
+    return;
+  }
+  if (mode === REST) {
+    restUntil = Date.now();  // skipping rest!
+    redrawApp();
+    return;
+  }
+}
+
+setWatch(buttonPress, BTN, { repeat: true, debounce: 25});
```

Comment:
I think you want to add `edge:"falling"` to the options passed here. Without it the app seemed to try and load my clock face when trying to skip the rest phase.

I'm not sure what debounce does here? removing it seems to have no effect on my watch.

Reply:
I set the debounce just so that the button would react to shorter presses. I have to long-press to go to my watch-face, do you maybe have another app that interferes here? I'll add the `edge: falling`, but I don't know how that will make a difference either TBQH... If that works better for you, that great :)

Reply:
Reacting on the falling edge makes sure there is no edge left for the next screen to react to.

Reply:
> do you maybe have another app that interferes here?

Yes it's the `fastreset` app that's the difference 🙂